### ISSUE TITLE
PROCESS CHANGE: Merging Client and Functions WG together

### ISSUE THREAD

**Issue by dsimansk** [knative/community#1554]
Things to include with your process proposal (delete this text):

* Expected benefits. Who gains the benefits? Why will they benefit?

Both Client WG and Functions WG represent a portion of Knative CLI. They are designed with different use cases in mind, but under the hood it's a terminal API. 

Currently, there's no high demand or interest in Client WG from both users and contributors POV. 

We have had a conversion wrt/ this proposal during previous TOC reviews, and as a part of community health discussion. 

* Expected costs. Who bears the costs? How heavy are they?

Less WG calls, in addition joint WG update. 

* Timeframe for implementation / rollout.

Week to sync WG calls and make announcements on channels, mailing list etc.

* Are you willing to drive the process, or is this a request for help?

Yes

Finally, I'd like to ask you all to vote for the name of this joint WG. @rhuss mentioned it would be great to keep `Functions` on top level rather wrap it in e.g. `CLI WG`. Please use thumb up on the comments below. Feel free to propose any suggestion that might be interesting.

New WG names proposals:

- `Functions & Client Working Group`
- `Functions & CLI Working Group`
- `CLI Working Group`

/cc @lkingland @rhuss 
/cc @knative/technical-oversight-committee 
/cc @knative/steering-committee

**Comment by dsimansk** [comment]
**Functions & CLI Working Group**

**Comment by dsimansk** [comment]
**CLI Working Group**

**Comment by cardil** [comment]
I think the name should be short and self-explanatory. Keeping "Functions" in WG name serves just historical purposes. New contributors would already understand they should look for Func work under CLI.

This isn't synonymous with the marketing aspect. We should keep the Func as one of the pillars of Knative, in prominent place on the website, docs, and our presentations. Those two are completely separate things to me.

**Comment by dprotaso** [comment]
+1 from me

**Comment by lkingland** [comment]
Looking forward to how this will help align our upcoming UX improvements!

**Comment by dsimansk** [comment]
A quick update after chatting with @lkingland on Functions WG call. The CLI will be spelled in plural form `CLIs`

**Functions & CLIs Working Group**

**Comment by dsimansk** [comment]
> I think the name should be short and self-explanatory. Keeping "Functions" in WG name serves just historical purposes. New contributors would already understand they should look for Func work under CLI.
> 
> This isn't synonymous with the marketing aspect. We should keep the Func as one of the pillars of Knative, in prominent place on the website, docs, and our presentations. Those two are completely separate things to me.

Indeed, it's a governance implementation detail. 
Roland's idea was aiming to keep "Functions" very prominent from all different angles. E.g. whenever you would go through community calendar looking at different WG calls/meetings - directly spotting "Functions... WG" rather than "CLI WG".

**Comment by psschwei** [comment]
+1 for merging, no opinion on name

**Comment by davidhadas** [comment]
+1

**Comment by rhuss** [comment]
The reason why I think that having "Functions" in the name of the Working group is to match our structure that considers the there pillars "Serving", "Eventing" and "Functions". This should be reflected in the organisational structure, too. See also the TOC of our docs:

![image](https://github.com/knative/community/assets/99080/002b3cdb-6041-4dfd-ae95-6b9b02f1f41f)

Here we might have still the issue that we refer to the Knative client as "Knative CLI". This documentation than also would need to be restructured, too.

**Comment by salaboy** [comment]
I agree with @rhuss Functions has become a fundamental part of the project.

**Comment by aliok** [comment]
+1 on merging from me.

I also like the name Functions better.

**Comment by aliok** [comment]
Can we have a +1 from leads of Client WG and Functions WG , TOC members and SC, if we want to go with this idea:
- Merge Client WG and Functions WG under "Functions & CLIs Working Group"

Summer is approaching, and we probably won't be able to vote this in a SC/TOC meeting (low meeting participation in summer).

**Comment by aliok** [comment]
cc @knative/steering-committee @knative/technical-oversight-committee @knative/client-wg-leads @knative/functions-wg-leads

**Comment by aliok** [comment]
+1 on "Merge Client WG and Functions WG under "Functions & CLIs Working Group"

**Comment by rhuss** [comment]
+1 for merging both WG, "Function & CLI Working Group" sound good to me (I would avoid the plural-s)

**Comment by dprotaso** [comment]
+1 on merging

**Comment by nainaz** [comment]
+1 for merging both WG. 
Name proposal: Function & CLI Working Group

**Comment by evankanderson** [comment]
I support a merge, but I think this is the TOC's call.  (It could be done async)

**Comment by psschwei** [comment]
I think we have TOC agreement on a merge, just need a final decision on the name. 

"Function & CLI Working Group" seems the popular choice, which works for me. Just need a consensus on  "CLI" vs. "CLIs"...

**Comment by dprotaso** [comment]
"Function & Client Working Group" might cover the basis that there could be many CLIs ?

**Comment by psschwei** [comment]
Same here, I'm ok with whatever the WG decides

**Comment by nrrso** [comment]
+1 for merging both WG.

Preferred Name: Function & CLI Working Group

**Comment by aliok** [comment]
@dsimansk Is there any blocker for this merger?

**Comment by dprotaso** [comment]
@dsimansk bump - can close this out by having the necessary updates to the https://github.com/knative/community/blob/main/working-groups/WORKING-GROUPS.md

 CNCF TOC is doing due diligence as part of graduation - so sooner the better
